subprojects/logging/src/main/java/org/gradle/internal/logging/events/LogEvent.java

```diff
@@ -29,7 +30,11 @@ public LogEvent(long timestamp, String category, LogLevel logLevel, String messa
     }
 
     public LogEvent(long timestamp, String category, LogLevel logLevel, String message, @Nullable Throwable throwable, @Nullable Object operationIdentifier) {
-        super(timestamp, category, logLevel, operationIdentifier);
+        this(timestamp, category, logLevel, message, throwable, operationIdentifier == null ? null : new CompactBuildOperationDescriptor(operationIdentifier, BuildOperationType.UNCATEGORIZED));
+    }
+
+    public LogEvent(long timestamp, String category, LogLevel logLevel, String message, @Nullable Throwable throwable, @Nullable CompactBuildOperationDescriptor buildOperationDescriptor) {
```

Comment:
The descriptor should go only in the start event, and only the build operation id should go in all the other events, to avoid sending the same information (eg its type) about a build operation many times, once per log event, instead of just once.